src/chains/vaults/chain.vaults.ts

```diff
@@ -23,41 +20,44 @@ export class ChainVaults {
     return this.cachedVaults;
   }
 
-  async getAllBy<T extends VaultCompoundModelTypes>(key: keyof IVaultCompoundModel, val: T) {
-    await this.#updateCachedVaults();
-    return this.cachedVaults.filter((vault) => vault[key] === val);
-  }
-
-  async getOneBy<T extends VaultCompoundModelTypes>(key: keyof IVaultCompoundModel, val: T) {
-    await this.#updateCachedVaults();
-    return this.cachedVaults.find((vault) => vault[key] === val);
+  async getVault(address: string) {
+    const vault = Object.fromEntries(this.cachedVaults.map((v) => [v.address, v]))[address];
+    if (!vault) {
+      throw new Error(`No vault exists with address ${address}`);
+    }
+    return vault;
   }
 
-  #isCacheTTLRelevant() {
-    if (!this.cacheLastUpd) return false;
-    return Date.now() - this.cacheLastUpd <= this.cacheTtl;
+  #shouldUpdate() {
+    if (!this.updatedAt) {
+      return true;
+    }
+    return Date.now() - this.updatedAt > this.cacheTtl;
   }
 
   async #updateCachedVaults() {
-    if (this.#isCacheTTLRelevant()) return;
-
-    const mapper = getDataMapper();
-    const query = mapper.query(
-      VaultCompoundModel,
-      { chain: this.network, isProduction: 1 },
-      { indexName: 'IndexVaultCompoundDataChainIsProd' },
-    );
-
-    try {
-      for await (const compoundVault of query) {
-        const cachedVaultIx = this.cachedVaults.findIndex((v) => v.address === compoundVault.address);
-        if (cachedVaultIx >= 0) this.cachedVaults[cachedVaultIx] = compoundVault;
-        else this.cachedVaults.push(compoundVault);
+    if (this.#shouldUpdate()) {
+      const mapper = getDataMapper();
+      const query = mapper.query(
+        VaultDefinitionModel,
+        { chain: this.network, isProduction: 1 },
```

Comment:
this should be right, coz we are constantly updating this value in `vault-compound-indexer` and if vault isnt on the list from onchain call, this means that it has been removed from production vaults, we ommit it on purpose. Or do u think, we want to see outdated vaults on staging?

Reply:
wouldn't that mean we can't have vault on just staging and not out on prod? If we remove a vault it would be removed on both prod/staging. I might not understand the full scope of how we handle staging/prod data in the whole system